charts/IndicatorDropdown.tsx

```diff
@@ -26,47 +76,256 @@ export class IndicatorDropdown extends React.Component<IndicatorDropdownProps> {
+                    <div className="indicator-current">
+                        <div className="indicator-info">
+                            <h1 className="indicator-title">
+                                {entity ? (
+                                    entity.title
+                                ) : (
+                                    <span className="none">
+                                        {entry === null
+                                            ? "Select an indicator"
+                                            : "Loading..."}
+                                    </span>
+                                )}
+                            </h1>
+                            {entity && entity.sourceDesc && (
+                                <div className="indicator-metadata">
+                                    {entity.sourceDesc}
+                                </div>
+                            )}
+                        </div>
+                        <span className="icon">
+                            <FontAwesomeIcon
+                                icon={this.isOpen ? faChevronUp : faChevronDown}
+                            />
+                        </span>
+                    </div>
+                </div>
+                {this.isOpen && (
+                    <IndicatorBrowser
+                        isOpen={this.isOpen}
+                        onChange={this.onChange}
+                        onClose={this.onClose}
+                        selectedId={entry?.entity?.id}
+                    />
+                )}
+            </React.Fragment>
+        )
     }
+}
 
-    @bind getValue(indicator: Indicator): string {
-        return `${indicator.id}`
+interface IndicatorBrowserProps {
+    isOpen: boolean
+    onChange: (indicator: Indicator) => void
+    onClose: () => void
+    selectedId: number | undefined
+}
+
+@observer
+export class IndicatorBrowser extends React.Component<IndicatorBrowserProps> {
+    static contextType = ExplorerViewContext
+    context!: ExplorerViewContextType
+
+    @observable.ref isLoading: boolean = true
+    @observable.ref indicators: Indicator[] = []
+    @observable.ref search: string = ""
+
+    async loadIndicators() {
+        const storeEntries = await this.context.store.indicators.getAll()
+        this.indicators = storeEntries
+            .map(entry => entry.entity)
+            .filter(entity => entity !== undefined) as Indicator[]
+        this.isLoading = false
     }
 
-    @bind getLabel(indicator: Indicator): string {
-        if (indicator.sourceDesc) {
-            return `${indicator.title}, ${indicator.sourceDesc}`
+    @computed get filteredIndicators(): Indicator[] {
+        if (this.search) {
+            return this.indicators.filter(
+                indicator =>
+                    (indicator.title || "")
+                        .toLocaleLowerCase()
+                        .indexOf(this.search.toLowerCase()) > -1
```

Comment:
You could use the more modern [`String.prototype.includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) here